dwave/system/temperatures.py

```diff
@@ -0,0 +1,596 @@
+            from pseudo-random samplers.
+
+        h_range (float, optional, default = [-1/6.1,1/6.1]):
+            Determines the range of external fields probed for temperature
+            inference. Default is based on a D-Wave Advantage processor, where
+            single-qubit freeze-out implies an effective temperature of 6.1 
+            (see :class:`~dwave.system.temperatures.freezeout_effective_temperature`).
+            The range should be chosen inversely proportional to the anticipated
+            temperature for statistical efficiency, and to accomodate precision 
+            and other nonidealities such as precision limitations.
+            
+        sampler_params (dict, optional):
+            Any additional non-defaulted sampler parameterization. If 
+            ``num_reads`` is a key, must be compatible with ``num_reads`` 
+            argument.
+
+        optimize_method (str, optional):
+            Optimize method used by SciPy ``root_scalar`` method. The default
+            method works well under default operation, 'bisect' can be 
+            numerically more stable when operated without defaults.
+
+        num_bootstrap_samples (int, optional, default=0):
+            Number of bootstrap samples to use for estimation of the
+            standard error. By default no bootstrapping is performed
+            and the standard error is defaulted to 0.
+
+    Returns:
+        Tuple[float, float]:
+            The effective temperature describing single qubit problems in an
+            external field, and a standard error (+/- 1 sigma). 
+            By default the confidence interval is set as 0.
+             
+    See also:
+
+        https://doi.org/10.3389/fict.2016.00023
+
+        https://www.jstor.org/stable/25464568
+    
+    Examples:
+       Draw samples from a :class:`~dwave.system.samplers.DWaveSampler`, and establish the temperature
+       
+       >>> from dwave.system.temperatures import fast_effective_temperature
+       >>> from dwave.system import DWaveSampler
+       >>> sampler = DWaveSampler()
+       >>> T, _ = fast_effective_temperature(sampler)
+       >>> print('Effective temperature at freeze-out is',T)    # doctest: +SKIP
+       0.21685104745347336
+    
+    See also:
+        The function :class:`~dwave.system.temperatures.freezeout_effective_temperature` 
+        may be used in combination with published device values to estimate single-qubit 
+        freeze-out, in approximate agreement with empirical estimates of this function.
+
+        https://doi.org/10.3389/fict.2016.00023
+
+        https://www.jstor.org/stable/25464568
+    
+    '''
+    
+    if sampler == None:
+        from dwave.system import DWaveSampler
+        sampler = DWaveSampler()
+        
+    if hasattr(sampler,'properties') and 'h_range' in sampler.properties:
+        if h_range[0] < sampler.properties['h_range'][0]:
+            h_range[0] = sampler.properties['h_range'][0]
```

Comment:
Note that this mutates the default value for `h_range` (effective in *next* function call as well), see comment about mutable default args above.